### PR TITLE
SProkopios/feature/293-add-feedback-card-component

### LINKED ISSUE #293: Add Feedback Card component
# ✨ **Feedback Popup Feature Proposal**

## **Feature Overview**

This proposal introduces a new feedback popup feature designed to provide users with multiple ways to share their feedback, offering a more dynamic and user-friendly interface. The component is intended for use across various parts of the application.

### **Feature Capabilities**

1. **Feedback through Icons**  
   Allow users to quickly share feedback by selecting emojies.
   
2. **Feedback through Text**  
   Provide a text input option for users who want to write detailed feedback.

3. **Redirect to Full Feedback Page**  
   Include a link for users to navigate to a dedicated feedback page for more comprehensive input.

---

## 🌟 **Benefits of the Feature**

- **Reusable Component**: Designed for use across multiple application pages.  
- **Improved User Experience**: Adapts to different user feedback preferences for seamless interaction.  
- **Increased Engagement**: Offers flexibility in feedback submission, encouraging more users to participate.  
- **Actionable Insights**: Provides varied feedback types for deeper analysis and improvements.

---

## 🛠️ **Proposed Implementation Steps**

1. **Develop a Modular Feedback Card Component**  
   - Build a flexible, reusable component to handle all feedback interactions.

2. **Integrate Storybook for Preview**  
   - Use Storybook to create and test various states of the feedback component.

3. **Support Internationalization (i18n)**  
   - Ensure the feature is accessible to a global audience by adding multi-language support.

4. **Implement Comprehensive Testing**  
   - Utilize React Testing Library for unit and integration tests to ensure reliability.

5. **Optional Directus Integration**  
   - Use POST methods to send feedback data to Directus, enabling centralized feedback storage and future scalability.

---

## 📎 **Additional Information**

- **Design Reference**:  
  [Figma Footer Plans](https://www.figma.com/design/qaRrNbozqOeESL9lQ2UDrM/Altzone?node-id=0-1&node-type=canvas&t=oECfc6YI5fetiUR3-0)

---

## **Visual Example**

Feedback Popup Example

---

This new feature aims to deliver a versatile and user-friendly feedback system, empowering users and enabling actionable insights for continuous improvement across the application.

### PR DESCRIPTION
**Pull request by SProkopios**
## 📄 **Pull Request Overview**

closes #293

## 🔧 **Changes Made**

1. Created feedback card component with test, story and multi-language support.

2. [Any refactoring or clean-up tasks]: Css-file looks bit messy with all the !important command. This is thanks to google fonts import, which messed the whole css-file for some reason.

---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected. ✅ Tested in storybook, test itself and in pages.
- **JSDoc**: I have added or updated JSDoc comments for all relevant code. ✅
- **Debugging**: No `console.log()` or other debugging statements are left. ✅
- **Clean Code**: Removed commented-out or unnecessary code. ✅
- **Tests**: Added new tests or updated existing ones for the changes made. ✅
- **Documentation**: Documentation has been updated (if applicable). ✅

---

## 📝 **Additional Information**

Provide any additional context or information that reviewers may need to know:

- **Screenshots**: [Include any screenshots or videos if the changes affect the UI]: -
- **Dependencies**: [Mention any new dependencies or breaking changes]: -

- **Known Issues**: [List any known issues or limitations]: Css-file bit messy because of google font import.

### CODE REVIEW

**Review by Jonroi — APPROVED** [review]
Everything is as planned and tests passed. Nice job!

**Review by leolabdev — CHANGES_REQUESTED** [review]
change feature name from "FeedbackCard" to "feedback"

features/feedback/ui (The card name remains the same.): 

image

Also every module should have API (index.ts), so make it:
features/feedback/index.ts

inside index.ts make FeebackCard export: 

export {default as FeedbackCard } from "path/to/FeedbackCard"


read more here: https://github.com/Alt-Org/Altzone-WebPages/wiki/6.-Project-Structure

File: frontend-next-migration/src/features/FeedbackCard/ui/FeedbackCard.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+/**
+ * FeedbackCard component for rendering a basic feedback card.
+ *
+ * @returns {JSX.Element} - The rendered feedback card component.
+ *
+ * @example
+ * <FeedbackCard/>
+ */
+
+export function FeedbackCard() {
```
Comment (leolabdev) [inline frontend-next-migration/src/features/FeedbackCard/ui/FeedbackCard.tsx:23]: change to export default

**Review by leolabdev — CHANGES_REQUESTED** [review]

**Review by leolabdev — CHANGES_REQUESTED** [review]
duplication ? 
image

File: frontend-next-migration/src/features/FeedbackCard/ui/FeedbackCard.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import React from 'react';
+import { FeedbackCard } from './FeedbackCard';
+
+export default {
+    title: 'features/FeedbackCard',
```
Comment (leolabdev) [inline frontend-next-migration/src/features/FeedbackCard/ui/FeedbackCard.stories.tsx:6]: features/feedback/FeedbackCard

**Review by leolabdev — CHANGES_REQUESTED** [review]

**Review by leolabdev — APPROVED** [review]

**Pull request merged**